sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -125,6 +125,17 @@ object ScalaReflection extends ScalaReflection {
     case _ => false
   }
 
+  def isValueClass(tpe: `Type`): Boolean = {
+    val notNull = !(tpe <:< localTypeOf[Null])
+    notNull && definedByConstructorParams(tpe) && tpe <:< localTypeOf[AnyVal]
```

Comment:
why a value class must be a `Product`? Is it because there is no way to get the value class field name and type?